helpers/metadata.ts

```diff
@@ -96,13 +99,22 @@ function compileSyncTable(syncTable: GenericSyncTable): PackSyncTable {
 
 function compileDefaultAuthenticationMetadata(
   authentication: Authentication | undefined,
+  networkDomains: string[] | undefined,
 ): AuthenticationMetadata | undefined {
   if (!authentication) {
     return;
   }
   if (authentication.type === AuthenticationType.None || authentication.type === AuthenticationType.Various) {
     return authentication;
   }
+
+  // User credentials must be explicitly pinned to exactly one domain so they can be invalidated if the pack
+  // owner changes the network domain. For simplicity and backwards compatibility, the authentication networkDomain
+  // can be omitted if there's exactly one call to setNetworkDomain elsewhere in the pack: we can safely assume
+  // authentication wants to be pinned to that same domain.
+  if (!authentication.networkDomain && networkDomains && networkDomains.length === 1) {
+    authentication.networkDomain = networkDomains[0];
```

Comment:
Just wondering about the value of populating this, if the `coda` code is always going to tolerate this being missing for single-domain packs. It raises the expectation that this will always be populated, but a year from now there could still be old builds where it isn't. Maybe that's fine and it's better to have most packs have this while still noting that old ones may not?

Reply:
For the moment the option that feels the most appealing to me is:
- Remove this logic in the SDK, with the intent that `coda` will support omitted auth networkDomain forever
- Centralize metadata normalization/transformation immediately after it's read in `coda`. Maybe put the version normalization function here in `packs-sdk` and have it return a type where the auth `networkDomain` is non-optional?
- Maybe start recording some kind of a version with the metadata?

Others options might be:
- Keep this logic with the hope that it makes a backfill simpler someday in the future
- Remove this logic and let the `coda` repo deal with the uncertainty about which SDK built some metadata in either a centralized or decentralized way

Reply:
I am leaning towards `Remove this logic in the SDK, with the intent that we'll support existing metadata with omitted auth networkDomain forever` unless we want to backfill the existing external metadata (doable but I don't feel worth for this change)

Reply:
Yeah I would be inclined to remove in the SDK and support existing metadata forever.

FWIW we do record the SDK version in the DB with every pack upload, which is meant to give us an indication of what feature set/structure was available to the pack at the time it was built. So it wouldn't be unreasonable to keep this logic here in the SDK and have the `coda` side only care about handling an undefined networkDomain if `sdkVersion < x.y.z`.

Reply:
Removed the metadata.ts changes